spec/gollum_git_repo_modification_spec.rb

```diff
@@ -0,0 +1,157 @@
+require 'spec_helper'
+require 'rspec/expectations'
+
+RSpec::Matchers.define :a_blob_named do |expected|
+  match do |actual|
+    actual.is_a?(Gollum::Git::Blob) && actual.name == expected
+  end
+end
+
+describe Gollum::Git::Repo do
+
+  after(:each) do
+    if parent_commit != repo.head.commit
+      repo.update_ref(repo.head.name, parent_commit.id)
+    end
+  end
+
+  let(:repo) { Gollum::Git::Repo.new(fixture('dot_bare_git'), :is_bare => true) }
+
+  let(:index) do
+    repo.index.read_tree(repo.head.commit.tree.id)
+    repo.index
+  end
+
+  let(:actor) { Gollum::Git::Actor.new("Tom Werner", "tom@example.com") }
+
+  let(:message) { 'Test commit' }
+
+  let(:parent_commit) { repo.head.commit }
+
+  def commit_index
+    index.commit(message, [parent_commit], actor, nil, repo.head.name)
+  end
+
+  describe "#head#commit" do
+
+    context "after committing" do
+
+      before(:each) do
+        index.add('Add.txt', 'Some data')
+        commit_index
```

Comment:
I am currently studying the init code for the repo-setup, but I do not understand how the temporary directory (created under `test/examples`) is deleted. Could you enlighten me?

My idea for setting up a temporary repository was more of using `Dir.mktmpdir` to copy the fixture repository:

```ruby
  before(:all) do
    @tmp = Dir.mktmpdir
    FileUtils.cp_r(File.join(fixture('dot_bare_git'), '.'), @tmp)
  end

  after(:all) do
    FileUtils.remove_entry(@tmp)
  end
```